### PR TITLE
Modify precommit script to "swiftlint" and "git clang-format" files.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Utilities/precommit.py [hunk 1]
```diff
@@ -29,6 +29,24 @@ def splitall(path):
...
+    
+    filename = os.path.basename(filepath)
+    if filename.startswith('.'):
+        return
+    file_ext = os.path.splitext(filename)[1]
+    if file_ext in ('.swift'):
+        env_copy = os.environ.copy()
+        env_copy["SCRIPT_INPUT_FILE_COUNT"] = "1"
+        env_copy["SCRIPT_INPUT_FILE_0"] = '%s' % ( short_filepath, )
+        lint_output = subprocess.check_output(['swiftlint', 'autocorrect', '--use-script-input-files'], env=env_copy)
```
Comment (charlesmchen) [inline Utilities/precommit.py:45]: First we run autocorrect...

File: Utilities/precommit.py [hunk 2]
```diff
@@ -29,6 +29,24 @@ def splitall(path):
...
+        print lint_output
+        lint_output = subprocess.check_output(['swiftlint', 'lint', '--use-script-input-files'], env=env_copy)
```
Comment (charlesmchen) [inline Utilities/precommit.py:47]: Then lint...

File: Utilities/precommit.py [hunk 3]
```diff
@@ -118,3 +133,6 @@ def process_if_appropriate(filepath):
         for filepath in filepaths:
             filepath = os.path.abspath(os.path.join(git_repo_path, filepath))
             process_if_appropriate(filepath)
+
+    print 'git clang-format...'
```
Comment (charlesmchen) [inline Utilities/precommit.py:137]: Lastly we run git clang-format.

**Comment by michaelkirk** [comment]
Love it. Ship it.

**Pull request merged**